src/libmongoc/tests/test-mongoc-write-commands.c

```diff
@@ -576,6 +576,132 @@ test_w0_legacy_insert_many (void)
    mock_server_destroy (server);
 }
 
+static void
+test_w0_legacy_update_one (void)
```

Comment:
OP_UPDATE previously had no test coverage. This method is based after the earlier OP_INSERT test, and I added another test below to ensure update/replace validation still occurs (via `mongoc-util.c` since we removed it from the legacy write methods).